### PR TITLE
sync with main branch + bump version to 4.3.4dev

### PR DESCRIPTION
**Pull request by boegel**

### CODE REVIEW

File: easybuild/tools/github.py [hunk 1]
```diff
@@ -311,6 +317,14 @@ def fetch_latest_commit_sha(repo, account, branch='master', github_user=None, to
...
     :return: latest SHA1
     """
+    if branch is None:
+        # use 'main' as default branch for 'easybuilders' organisation,
+        # otherwise use 'master'
+        if account == GITHUB_EB_MAIN:
+            branch = 'main'
+        else:
+            branch = 'master'
+
```
Comment (akesandgren) [inline easybuild/tools/github.py:327]: Should probably split out to a separate func since it's used 3 times.

Comment (boegel) [reply easybuild/tools/github.py:327]: That should be done in a separate PR, I can take care of that after merging this one...

Comment (akesandgren) [reply easybuild/tools/github.py:327]: ok.

**Review by akesandgren — APPROVED** [review]
LGTM

**Comment by akesandgren** [comment]
Going in, thanks @boegel!

**Pull request merged**